catalog_integrations_api_tools/parser.py

```diff
@@ -0,0 +1,269 @@
+        """
+
+        # Rename file columns based on Catalog API parameter names
+        df.rename(
+            columns={
+                "SKU": "sku",
+                "BRAND": "brand",
+                "NAME": "name",
+                "BARCODES": "barcodes",
+                "URL": "url",
+                "BASE_UNIT": "base_unit",
+                "CONTENT_UNIT": "content_unit",
+                "DESCRIPTION": "description",
+                "PACKAGE": "package",
+                "CATEGORIES": "categories",
+                "CONTENT_UNIT_SIZE": "content_unit_size",
+                "UNIT_CONVERSION_RATE": "unit_conversion_rate",
+                "MAIN_IMAGE_URL": "main_image_url",
+                "IMAGE_URLS": "image_urls",
+                "WEIGHT": "weight",
+                "VARIABLE_WEIGHT": "has_variable_weight",
+                "TAX_PERCENTAGE": "tax",
+                "TAGS": "tags",
+            },
+            inplace=True,
+        )
+        # Validate data and raise errors if needed
+        Parser._validate_products(df)
+        # Drop unnecessary columns
+        df.drop(labels=["REFERENCE", "NUTRITIONAL_INFO"], inplace=True, axis=1)
+        # Add category uuid column if it is not already there
+        if "category_uuid" not in df.columns:
+            df["category_uuid"] = nan
+        # Reformat fields to be in list format
+        df["barcodes"] = df["barcodes"].apply(Parser._jsonify_field, args=(",",))
+        df["categories"] = df["categories"].apply(Parser._jsonify_field, args=("/",))
+        df["tags"] = df["tags"].apply(Parser._jsonify_field, args=(",",))
+        df["image_urls"] = df["image_urls"].apply(Parser._jsonify_field, args=(",",))
+
+        return df
+
+    @staticmethod
+    def _jsonify_field(field, separator):
+        if not isinstance(field, float):  # Check that the value is not nan
+            return json.dumps([i.strip() for i in str(field).split(separator)])
+        return field
+
+    @staticmethod
+    def parse_stock_prices(df: pd.DataFrame, currency: str,) -> None:
+        """
+        Parse stock prices file to the correct format for catalog API.
+        The dataframe is modified in place.
+
+        Args:
+            df (pd.DataFrame): stock-prices df.
+            currency (str): Currency of prices.
+
+        Raises:
+            ValueError: If required fields are missing.
+
+        Returns:
+            pd.DataFrame: Parsed df.
+        """
+
+        df.rename(
+            columns={"SKU": "sku", "STOCK": "stock", "PRICE": "price"}, inplace=True
+        )
+        # Validate values
+        Parser._validate_stock_prices(df)
+
+        # Scale price so that it is in zero denomination format
+        df["price"] *= Parser.currencies[currency]
+        # Add missing columns
+        df["currency"] = currency
+        df["status"] = df["stock"].apply(
+            lambda x: "AVAILABLE" if x > 0 else "OUT_OF_STOCK"
+        )
```

Comment:
This assumption might not be correct for al integrations.

Reply:
Hmm yes.
What do you suggest?

Reply:
we get the stock_threshold from the integrator as a dictionary.
example --> {"AVAILABLE": 2, "OUT_OF_STOCK": 1, "UNAVAILABLE": 0}
```suggestion
        df["status"] = df["stock"].apply(get_stock_status)
```

Reply:
`def get_stock_status(stock):
    if stock >= self.stock_threshold["AVAILABLE"]:
        return "AVAILABLE"
    elif stock >= self.stock_threshild["OUT_OF_STOCK"]:
        return "OUT_OF_STOCK":
    else:
        return "UNAVAILABLE"`